app/services/case_remove_sar_deadline_extension_service.rb

```diff
@@ -0,0 +1,28 @@
+class CaseRemoveSARDeadlineExtensionService
+  attr_reader :result, :error
+
+  def initialize(user, kase)
+    @user = user
+    @case = kase
+    @result = :incomplete
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      @case.state_machine.remove_sar_deadline_extension!(
+        acting_user: @user,
+        acting_team: @user.team_for_case(@case)
+      )
+
+      @case.reset_deadline!
+      @result = :ok
+    end
+    @result
+  rescue => err
+    Rails.logger.error err.to_s
+    Rails.logger.error err.backtrace.join("\n\t")
+    @error = err
+    @result = :error
+    raise err
```

Comment:
Given my comment above, I think this might be a bug...?

Reply:
Discussed with @starswan - leave as is for now as the handling of errors is a system wide issue requiring a broader discussion